controller/src/main/java/cern/jarrace/controller/manager/impl/InMemoryAgentContainerManager.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * © Copyright 2016 CERN. This software is distributed under the terms of the Apache License Version 2.0, copied
+ * verbatim in the file “COPYING”. In applying this licence, CERN does not waive the privileges and immunities granted
+ * to it by virtue of its status as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+package cern.jarrace.controller.manager.impl;
+
+import cern.jarrace.commons.domain.AgentContainer;
+import cern.jarrace.controller.manager.AgentContainerManager;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Implementation of {@link AgentContainerManager} that stores the information in an "in memory" set
+ *
+ * @author tiagomr
+ */
+public class InMemoryAgentContainerManager implements AgentContainerManager {
+
+    private final Set<AgentContainer> agentContainers = Collections.synchronizedSet(new HashSet<>());
+
+    @Override
+    public AgentContainer findAgentContainer(String containerName) {
+        if (containerName == null || containerName.isEmpty()) {
+            throw new IllegalArgumentException("Name of the container cannot be null nor empty");
+        }
+        synchronized (agentContainers) {
+            Optional<AgentContainer> toReturn = agentContainers.stream().filter(agentContainer -> {
```

Comment:
Yes, but the whole purpose of this class is the lookup ... Where a map seems to be appropriate. And if you do not expose it, nobody cares, because it is always consisten.